src/invocation/InvocationService.ts

```diff
@@ -213,8 +225,35 @@ export class Invocation {
     }
 
     complete(clientMessage: ClientMessage): void {
-        this.deferred.resolve(clientMessage);
-        this.invocationService.deregisterInvocation(this.request.getCorrelationId());
+        /**
+         * The following is a part of eager deserialization that is needed for compact serialization. In
+         * eager deserialization invocations have handlers that are called in {@link complete}.
+         * This invocation handler usually includes toObject calls and as seen below it is put
+         * into a try/catch block because {@link CompactStreamSerializer} might throw {@link SchemaNotFoundError}.
+         * If it throws, we fetch the required schema and try to call handler again. If the handler is successful,
+         * we complete the invocation. If another exception is thrown we complete the invocation with an error.
+         */
+        try {
+            const result = this.handler(clientMessage);
+            this.deferred.resolve(result);
```

Comment:
That's not a problem, you can resolve a promise after rejecting it or visa versa. Or you can resolve it many times. Always the first action counts.

Reply:
> Always the first action counts.

Then that code block is wrong, isn't it? The promise is first resolved then if there's an error it is rejected. So wouldn't it be resolved OK (since that's the first action) although there's an error?

Reply:
The only thing we retry `complete()` is the case of SchemaNotReplicatedError thrown. And in that case we only resolve or reject the promise